packages/server/src/automations/tests/utilities/AutomationTestBuilder.ts

```diff
@@ -43,69 +46,55 @@ type TriggerOutputs =
   | CronTriggerOutputs
   | undefined
 
-class AutomationBuilder {
-  private automationConfig: Automation = {
-    name: "",
-    definition: {
-      steps: [],
-      trigger: {} as AutomationTrigger,
-    },
-    type: "automation",
-    appId: setup.getConfig().getAppId(),
-  }
-  private config: TestConfiguration = setup.getConfig()
-  private triggerOutputs: TriggerOutputs
-  private triggerSet: boolean = false
-
-  constructor(options: { name?: string } = {}) {
-    this.automationConfig.name = options.name || `Test Automation ${uuidv4()}`
+type BranchConfig = {
+  [key: string]: {
+    steps: (stepBuilder: StepBuilder) => StepBuilder
+    condition: SearchFilters
   }
+}
 
-  // TRIGGERS
-  rowSaved(inputs: RowCreatedTriggerInputs, outputs: RowCreatedTriggerOutputs) {
-    this.triggerOutputs = outputs
-    return this.trigger(
-      TRIGGER_DEFINITIONS.ROW_SAVED,
-      AutomationTriggerStepId.ROW_SAVED,
-      inputs,
-      outputs
-    )
-  }
+class BaseStepBuilder {
+  protected steps: AutomationStep[] = []
 
-  rowUpdated(
-    inputs: RowUpdatedTriggerInputs,
-    outputs: RowUpdatedTriggerOutputs
-  ) {
-    this.triggerOutputs = outputs
-    return this.trigger(
-      TRIGGER_DEFINITIONS.ROW_UPDATED,
-      AutomationTriggerStepId.ROW_UPDATED,
-      inputs,
-      outputs
-    )
+  protected step<TStep extends AutomationActionStepId>(
+    stepId: TStep,
+    stepSchema: Omit<AutomationStep, "id" | "stepId" | "inputs">,
+    inputs: AutomationStepInputs<TStep>
+  ): this {
+    this.steps.push({
+      ...stepSchema,
+      inputs: inputs as any,
+      id: uuidv4(),
+      stepId,
+    })
+    return this
   }
 
-  rowDeleted(
-    inputs: RowDeletedTriggerInputs,
-    outputs: RowDeletedTriggerOutputs
-  ) {
-    this.triggerOutputs = outputs
-    return this.trigger(
-      TRIGGER_DEFINITIONS.ROW_DELETED,
-      AutomationTriggerStepId.ROW_DELETED,
-      inputs,
-      outputs
-    )
-  }
+  branch(branchConfig: BranchConfig): this {
```

Comment:
I think the branch step should return a more limited set, rather than `this` - since you can't add steps after the branch we can just return the `run` function.